test-tests/shared/src/test/scala/zio/test/environment/EnvironmentSpec.scala

```diff
@@ -0,0 +1,62 @@
+package zio.test.environment
+
+import java.util.concurrent.TimeUnit
+
+import zio._
+import zio.duration._
+import zio.test.Assertion._
+import zio.test._
+
+object EnvironmentSpec
+    extends ZIOBaseSpec(
+      suite("EnvironmentSpec")(
+        testM("Clock returns time when it is set") {
+          for {
+            _    <- TestClock.setTime(1.millis)
+            time <- clock.currentTime(TimeUnit.MILLISECONDS)
+          } yield assert(time, equalTo(1L))
+        },
+        testM("Console writes line to output") {
+          for {
+            _      <- console.putStrLn("First line")
+            _      <- console.putStrLn("Second line")
+            output <- TestConsole.output
+          } yield assert(output, equalTo(Vector("First line\n", "Second line\n")))
+        },
+        testM("Console reads line from input") {
+          for {
+            _      <- TestConsole.feedLines("Input 1", "Input 2")
+            input1 <- console.getStrLn
+            input2 <- console.getStrLn
+          } yield {
+            assert(input1, equalTo("Input 1"))
+            assert(input2, equalTo("Input 2"))
+          }
+        },
+        testM("Random returns next pseudorandom integer") {
```

Comment:
We lost something with this test. In the previous version we called `provideManaged(TestEnvironment.Value`) on each of the effects. That is important because what we are testing for in this assertion, which admittedly isn't labeled as clearly as it should be, is that the random number generators for different copies of the `TestEnvironment` are seeded with different seeds. If we don't do this both calls will use the same copy of the `TestEnvironment` so we won't be testing the same thing.

Reply:
Ah, thought I was being clever by figuring out I could use the TestEnvironment provided generator - I'll update the test description

Reply:
I think I'll actually leave this test as is and create another one to check different copies of `TestEnvironment` are seeded with different seeds. I think it's important that we test the random number generator isn't just returning the same number for successive calls...

Reply:
Sounds good. We also have a ton of additional tests in `RandomSpec`.